Compilation error when compiling tracing-subscriber 0.3 with time but without local-time

## Bug Report

### Version

```
│   ├── tracing v0.1.26
│   │   │   ├── tracing-attributes v0.1.15 (proc-macro)
│   │   │   └── tracing-core v0.1.21
│   │   └── tracing-subscriber v0.3.0
│   │       ├── tracing v0.1.26 (*)
│   │       └── tracing-core v0.1.21 (*)
│   │   └── tracing v0.1.26 (*)
│   ├── tracing v0.1.26 (*)
│   ├── tracing-futures v0.2.5
│   │   └── tracing v0.1.26 (*)
│   ├── tracing-subscriber v0.3.0 (*)
├── tracing v0.1.26 (*)
└── tracing-subscriber v0.3.0 (*)
├── tracing v0.1.26 (*)
├── tracing-subscriber v0.3.0 (*)
```

### Platform

Darwin djc-2019 20.6.0 Darwin Kernel Version 20.6.0: Mon Aug 30 06:12:21 PDT 2021; root:xnu-7195.141.6~3/RELEASE_X86_64 x86_64

### Crates

tracing-subscriber

### Description

Upgrading from tracing-subscriber 0.2 to 0.3, I found that just replacing the chrono feature with time did not work:

```rust
error[E0432]: unresolved import `time_crate::LocalTime`
  --> /Users/djc/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-subscriber-0.3.0/src/fmt/time/mod.rs:13:22
   |
13 | pub use time_crate::{LocalTime, UtcTime};
   |                      ^^^^^^^^^ no `LocalTime` in `fmt::time::time_crate`
```

--- a/tracing-subscriber/src/fmt/time/mod.rs
+++ b/tracing-subscriber/src/fmt/time/mod.rs
@@ -6,11 +6,14 @@ use std::time::Instant;
 mod datetime;
 
 #[cfg(feature = "time")]
-#[cfg_attr(docsrs, doc(cfg(feature = "time")))]
 mod time_crate;
 #[cfg(feature = "time")]
 #[cfg_attr(docsrs, doc(cfg(feature = "time")))]
-pub use time_crate::{LocalTime, UtcTime};
+pub use time_crate::UtcTime;
+
+#[cfg(feature = "local-time")]
+#[cfg_attr(docsrs, doc(cfg(feature = "local-time")))]
+pub use time_crate::LocalTime;
 
 /// A type that can measure and format the current time.
 ///

